src/utils/formatCurrencyAmount.ts

```diff
@@ -1,41 +1,17 @@
-import { Currency, CurrencyAmount, Fraction, Price } from '@uniswap/sdk-core'
-import { DEFAULT_LOCALE, SupportedLocale } from 'constants/locales'
-import JSBI from 'jsbi'
-import formatLocaleNumber from 'utils/formatLocaleNumber'
+import { Currency, CurrencyAmount, Price } from '@uniswap/sdk-core'
 
-export function formatCurrencyAmount(
-  amount: CurrencyAmount<Currency> | undefined,
-  sigFigs: number,
-  locale: SupportedLocale = DEFAULT_LOCALE,
-  fixedDecimals?: number
-): string {
-  if (!amount) {
-    return '-'
-  }
+import {
+  currencyAmountToPreciseFloat,
+  formatDollar,
+  formatTransactionAmount,
+  priceToPreciseFloat,
+} from './formatNumbers'
 
-  if (JSBI.equal(amount.quotient, JSBI.BigInt(0))) {
-    return '0'
-  }
-
-  if (amount.divide(amount.decimalScale).lessThan(new Fraction(1, 100000))) {
-    return `<${formatLocaleNumber({ number: 0.00001, locale, sigFigs, fixedDecimals })}`
-  }
-
-  return formatLocaleNumber({ number: amount, locale, sigFigs, fixedDecimals })
+export function formatCurrencyAmount(amount: CurrencyAmount<Currency> | undefined, isUsdPrice = false): string {
+  const currencyAmountNumber = currencyAmountToPreciseFloat(amount)
+  return isUsdPrice ? formatDollar({ num: currencyAmountNumber }) : formatTransactionAmount(currencyAmountNumber)
```

Comment:
What will undefined return? Please include in a docstring (for all these fn's).

Reply:
good point, returned empty string. added documentation.